### PR TITLE
Fix writes out of order

### PR DESCRIPTION
**Pull request by JonasHallFnx**
Continue to enqueue writes until queue is drained, to avoid that writes comes out of order when writes are triggered from both the event loop and other threads.

Solves issue #605

### CODE REVIEW

**Comment by JonasHallFnx** [comment]
Any maintainers available for review? @jamesgorman2 ?

**Review by jamesgorman2 — APPROVED** [review]
LGTM

There might still be a race around the first off event loop write (a write(inEL) concurrent with the _first_ write(!inEL)), but that already exists and isn't introduced by this PR.

@TurboProgramming I'm approving for now, but can you can have think and I'll merge or hold off after I do the same and get feedback from you

**Comment by JonasHallFnx** [comment]
@jamesgorman2 any chance we can get this merged?

**Pull request merged**
